matlab_simulation/code v1.0/ME597-7-NonlinearSteering/car_trajectory.m

```diff
@@ -0,0 +1,107 @@
+%% Trajectory tracking
+
+% Fixed vehicle parameters
+velocity = 5; % Speed
+delta_max = 25*pi/180; % max steering angle
+k = 2.5; % Gain
+robot_length = 1; % Car length
+
+% Desired trajectory as a plot of points. The plot of points defines
+% multiple line segments all joined together. The vehicle will first follow
+% along the first line segment. After it is has driven past the length of
+% said line segment, it defines a new line segment to follow by taking the
+% next point.
+
+
+traj_points = [0, 0;
+               10,0;
+               10,20
+               -10, 20];
+
+traj_point_counter = 1; % Keep track of where we are in the trajectory
+           
+% Initial conditions in [x y heading]
+x0 = [0 5 0]; 
+
+% Simulation time
+Tmax = 15;  % End point
+dt =0.01; % Time step
+T = 0:dt:Tmax; % Time vector
+
+% Simulation setup
+xd = zeros(length(T)-1,3); % Derivative of state ([edot psidot])
+x = zeros(length(T),3);  % State ([e psi] 
+x(1,:) = x0; % Initial condition
+delta = zeros(length(T),1); % Steering angles
+
+figure(1);clf; hold on;
+axis([-10 10 -10 10]);
+for i=1:length(T)-1
+    % The desired trajectory is a line segment consisting of 2 points from
+    % the desired trajectory
+    end_point = traj_points(traj_point_counter+1, :);
+    beginning = traj_points(traj_point_counter, :);
```

Comment:
It would be nice to have a consistent naming convention for the beginning and end points, so on a quick glance it is clear they are the same type of thing. Something like `beginning_point`, `beg_point`, or `start_point` would be better.